build_tools/circle/build_doc.sh

```diff
@@ -57,9 +57,12 @@ get_build_type() {
         echo QUICK BUILD: no changed filenames for $git_range
         return
     fi
-    if echo "$filenames" | grep -q -e ^examples/
+    changed_examples=$(echo "$filenames" | grep -e ^examples/)
+    if [[ -n "$changed_examples" ]]
     then
-        echo BUILD: detected examples/ filename modified in $git_range: $(echo "$filenames" | grep -e ^examples/ | head -n1)
+        echo BUILD: detected examples/ filename modified in $git_range: $changed_examples
+        pattern=$(echo "$changed_examples" | paste -sd '|')
+        echo "$pattern"
```

Comment:
Deserves a comment

Reply:
Happy to add a comment but maybe you can tell about what exactly? Maybe that $pattern is a regex that decides which examples will be run? Note that I have a comment like this when `build_type` is called below:

```python
    # pattern for examples to run is the last line of output
```

but maybe this is not enough?

Reply:
I just meant to explain why it's printing